setup.py

```diff
@@ -63,6 +43,89 @@ class CustomBuild(build):
     ]
 
 
+def find_glpk_header():
+    # If a path is provded by the environment, expect the GLPK header there.
+    given_path = os.environ.get(ENV_GLPK_HEADER_PATH, None)
+    if given_path:
+        if not os.path.isdir(given_path):
+            raise ValueError(
+                'The environment variable {}="{}" is set but does not point to a directory.'
+                .format(ENV_GLPK_HEADER_PATH, given_path)
+            )
+
+        header = os.path.join(given_path, GLPK_HEADER_NAME)
+
+        if not os.path.isfile(header):
+            raise ValueError(
+                'The environment variable {}="{}" is set to a directory that does not contain {}.'
+                .format(ENV_GLPK_HEADER_PATH, given_path, GLPK_HEADER_NAME)
+            )
+
+        return os.path.abspath(given_path)
+
+    # Look for a drop-in header file next.
+    if os.path.isfile(GLPK_HEADER_NAME):
+        return os.getcwd()
+
+    include_dirs = list(INCLUDE_DIRS)
+
+    # If glpsol is found, look for an include directory in its vicinity.
+    glpsol = which(GLPSOL_BINARY_NAME)
+    if glpsol:
+        glpsol_path = os.path.dirname(glpsol)
+        glpsol_root = os.path.dirname(glpsol_path)
+        glpsol_include_path = os.path.join(glpsol_root, "include")
+        include_dirs.insert(0, glpsol_include_path)
+
+    # Look at common places.
+    for path in include_dirs:
+        if os.path.isdir(path):
+            header = os.path.join(path, GLPK_HEADER_NAME)
+
+            if os.path.isfile(header):
+                return os.path.abspath(path)
+
+    raise FileNotFoundError('Failed to locate {}.'.format(GLPK_HEADER_NAME))
+
+
+# Warn users installing via pip that this is a source build.
+print(
+    '='*30,
+    'BUILDING SWIGLPK FROM SOURCE.',
+    'If you are installing SWIGLPK via pip, this means that no wheels are'
+    ' offered for your platform or Python version yet.',
+    'This can be the case if you adopt a new Python version early.',
+    'For a source build, you need both GLPK and SWIG to be installed!',
```

Comment:
Also needs GMP on linux and mac which is an optional GLPK dependency.

Reply:
Will add.